File: fstp-python/com/purefun/fstp/tushare/bo/otw/StockBasicInfoBO_OTW.py
from core.common.ICommon_OTW import ICommon_OTW
from com.purefun.fstp.tushare.bo.pro import StockBasicInfoBO_pb2
from com.purefun.fstp.tushare.bo.model.StockBasicInfoBO import StockBasicInfoBO

class StockBasicInfoBO_OTW(ICommon_OTW):

    def __init__(self, byteMsg = None):
        self._bo_pro = StockBasicInfoBO_pb2.StockBasicInfoBO()
        self._bo = StockBasicInfoBO()

        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()

    def __setDataFromBO(self):
        self._bo_pro.stockcode = self._bo.stockcode
        self._bo_pro.stockname = self._bo.stockname
        self._bo_pro.industry = self._bo.industry
        self._bo_pro.area = self._bo.area
        self._bo_pro.pe = self._bo.pe
        self._bo_pro.offer_shares = self._bo.offer_shares
        self._bo_pro.total_shares = self._bo.total_shares
        self._bo_pro.totalAssets = self._bo.totalAssets
        self._bo_pro.liquidAssets = self._bo.liquidAssets
        self._bo_pro.fixedAssets = self._bo.fixedAssets
        self._bo_pro.reserved = self._bo.reserved
        self._bo_pro.reservedPerShare = self._bo.reservedPerShare
        self._bo_pro.esp = self._bo.esp
        self._bo_pro.bvps = self._bo.bvps
        self._bo_pro.pb = self._bo.pb
        self._bo_pro.list_date = self._bo.list_date
        self._bo_pro.no_dividend = self._bo.no_dividend
        self._bo_pro.perundp = self._bo.perundp
        self._bo_pro.rev_per = self._bo.rev_per
        self._bo_pro.profit = self._bo.profit
        self._bo_pro.gross_profit = self._bo.gross_profit
        self._bo_pro.net_profit = self._bo.net_profit
        self._bo_pro.holder = self._bo.holder
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.destination = self._bo.destination

    def __setDataFromPB(self):
        self._bo.stockcode = self._bo_pro.stockcode
        self._bo.stockname = self._bo_pro.stockname
        self._bo.industry = self._bo_pro.industry
        self._bo.area = self._bo_pro.area
        self._bo.pe = self._bo_pro.pe
        self._bo.offer_shares = self._bo_pro.offer_shares
        self._bo.total_shares = self._bo_pro.total_shares
        self._bo.totalAssets = self._bo_pro.totalAssets
        self._bo.liquidAssets = self._bo_pro.liquidAssets
        self._bo.fixedAssets = self._bo_pro.fixedAssets
        self._bo.reserved = self._bo_pro.reserved
        self._bo.reservedPerShare = self._bo_pro.reservedPerShare
        self._bo.esp = self._bo_pro.esp
        self._bo.bvps = self._bo_pro.bvps
        self._bo.pb = self._bo_pro.pb
        self._bo.list_date = self._bo_pro.list_date
        self._bo.no_dividend = self._bo_pro.no_dividend
        self._bo.perundp = self._bo_pro.perundp
        self._bo.rev_per = self._bo_pro.rev_per
        self._bo.profit = self._bo_pro.profit
        self._bo.gross_profit = self._bo_pro.gross_profit
        self._bo.net_profit = self._bo_pro.net_profit
        self._bo.holder = self._bo_pro.holder
        self._bo.uuid = self._bo_pro.uuid
        self._bo.boid = self._bo_pro.boid
        self._bo.destination = self._bo_pro.destination

    def getBO(self):
        return self._bo

    def getProBO(self):
        return self._bo_pro

    def getStockcode(self):
        return self._bo.stockcode

    def setStockcode(self, stockcode):
        self._bo.stockcode = stockcode
        self._bo_pro.stockcode = stockcode

    def getStockname(self):
        return self._bo.stockname

    def setStockname(self, stockname):
        self._bo.stockname = stockname
        self._bo_pro.stockname = stockname

    def getIndustry(self):
        return self._bo.industry

    def setIndustry(self, industry):
        self._bo.industry = industry
        self._bo_pro.industry = industry

    def getArea(self):
        return self._bo.area

    def setArea(self, area):
        self._bo.area = area
        self._bo_pro.area = area

    def getPe(self):
        return self._bo.pe

    def setPe(self, pe):
        self._bo.pe = pe
        self._bo_pro.pe = pe

    def getOffer_shares(self):
        return self._bo.offer_shares

    def setOffer_shares(self, offer_shares):
        self._bo.offer_shares = offer_shares
        self._bo_pro.offer_shares = offer_shares

    def getTotal_shares(self):
        return self._bo.total_shares

    def setTotal_shares(self, total_shares):
        self._bo.total_shares = total_shares
        self._bo_pro.total_shares = total_shares

    def getTotalAssets(self):
        return self._bo.totalAssets

    def setTotalAssets(self, totalAssets):
        self._bo.totalAssets = totalAssets
        self._bo_pro.totalAssets = totalAssets

    def getLiquidAssets(self):
        return self._bo.liquidAssets

    def setLiquidAssets(self, liquidAssets):
        self._bo.liquidAssets = liquidAssets
        self._bo_pro.liquidAssets = liquidAssets

    def getFixedAssets(self):
        return self._bo.fixedAssets

    def setFixedAssets(self, fixedAssets):
        self._bo.fixedAssets = fixedAssets
        self._bo_pro.fixedAssets = fixedAssets

    def getReserved(self):
        return self._bo.reserved

    def setReserved(self, reserved):
        self._bo.reserved = reserved
        self._bo_pro.reserved = reserved

    def getReservedPerShare(self):
        return self._bo.reservedPerShare

    def setReservedPerShare(self, reservedPerShare):
        self._bo.reservedPerShare = reservedPerShare
        self._bo_pro.reservedPerShare = reservedPerShare

    def getEsp(self):
        return self._bo.esp

    def setEsp(self, esp):
        self._bo.esp = esp
        self._bo_pro.esp = esp

    def getBvps(self):
        return self._bo.bvps

    def setBvps(self, bvps):
        self._bo.bvps = bvps
        self._bo_pro.bvps = bvps

    def getPb(self):
        return self._bo.pb

    def setPb(self, pb):
        self._bo.pb = pb
        self._bo_pro.pb = pb

    def getList_date(self):
        return self._bo.list_date

    def setList_date(self, list_date):
        self._bo.list_date = list_date
        self._bo_pro.list_date = list_date

    def getNo_dividend(self):
        return self._bo.no_dividend

    def setNo_dividend(self, no_dividend):
        self._bo.no_dividend = no_dividend
        self._bo_pro.no_dividend = no_dividend

    def getPerundp(self):
        return self._bo.perundp

    def setPerundp(self, perundp):
        self._bo.perundp = perundp
        self._bo_pro.perundp = perundp

    def getRev_per(self):
        return self._bo.rev_per

    def setRev_per(self, rev_per):
        self._bo.rev_per = rev_per
        self._bo_pro.rev_per = rev_per

    def getProfit(self):
        return self._bo.profit

    def setProfit(self, profit):
        self._bo.profit = profit
        self._bo_pro.profit = profit

    def getGross_profit(self):
        return self._bo.gross_profit

    def setGross_profit(self, gross_profit):
        self._bo.gross_profit = gross_profit
        self._bo_pro.gross_profit = gross_profit

    def getNet_profit(self):
        return self._bo.net_profit

    def setNet_profit(self, net_profit):
        self._bo.net_profit = net_profit
        self._bo_pro.net_profit = net_profit

    def getHolder(self):
        return self._bo.holder

    def setHolder(self, holder):
        self._bo.holder = holder
        self._bo_pro.holder = holder

    def getUuid(self):
        return self._bo.uuid

    def setUuid(self, uuid):
        self._bo.uuid = uuid
        self._bo_pro.uuid = uuid

    def getBoid(self):
        return self._bo.boid

    def setBoid(self, boid):
        self._bo.boid = boid
        self._bo_pro.boid = boid

    def getDestination(self):
        return self._bo.destination

    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination

    def toString(self):
        return "StockBasicInfoBO_OTW ["+"stockcode = " + self.getStockcode() +"," +"stockname = " + self.getStockname() +"," +"industry = " + self.getIndustry() +"," +"area = " + self.getArea() +"," +"pe = " + str(self.getPe()) +"," +"offer_shares = " + str(self.getOffer_shares()) +"," +"total_shares = " + str(self.getTotal_shares()) +"," +"totalAssets = " + str(self.getTotalAssets()) +"," +"liquidAssets = " + str(self.getLiquidAssets()) +"," +"fixedAssets = " + str(self.getFixedAssets()) +"," +"reserved = " + str(self.getReserved()) +"," +"reservedPerShare = " + str(self.getReservedPerShare()) +"," +"esp = " + str(self.getEsp()) +"," +"bvps = " + str(self.getBvps()) +"," +"pb = " + str(self.getPb()) +"," +"list_date = " + self.getList_date() +"," +"no_dividend = " + str(self.getNo_dividend()) +"," +"perundp = " + str(self.getPerundp()) +"," +"rev_per = " + str(self.getRev_per()) +"," +"profit = " + str(self.getProfit()) +"," +"gross_profit = " + str(self.getGross_profit()) +"," +"net_profit = " + str(self.getNet_profit()) +"," +"holder = " + str(self.getHolder()) +"," +"uuid = " + self.getUuid() +"," +"boid = " + str(self.getBoid()) +"," +"destination = " + self.getDestination() +"," +"]"

File: fstp-core-bo/resource/com/purefun/fstp/tushare/bo/otw/StockBasicInfoBO_OTW.java
package com.purefun.fstp.tushare.bo.otw;

import com.purefun.fstp.tushare.bo.StockBasicInfoBO;
import com.purefun.fstp.tushare.bo.pro.StockBasicInfoBO_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fstp.core.bo.commom.ICommom_OTW;
import com.google.protobuf.Any;

public class StockBasicInfoBO_OTW implements ICommom_OTW {
    StockBasicInfoBO_PRO.StockBasicInfoBO.Builder builder = null;
    StockBasicInfoBO bo = null;

    public StockBasicInfoBO_OTW() {
        builder = StockBasicInfoBO_PRO.StockBasicInfoBO.newBuilder();
        bo= new StockBasicInfoBO();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public StockBasicInfoBO_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = StockBasicInfoBO_PRO.StockBasicInfoBO.newBuilder();
        bo= new StockBasicInfoBO();
        StockBasicInfoBO_PRO.StockBasicInfoBO receive = StockBasicInfoBO_PRO.StockBasicInfoBO.parseFrom(message);
        setStockcode(receive.getStockcode());
        setStockname(receive.getStockname());
        setIndustry(receive.getIndustry());
        setArea(receive.getArea());
        setPe(receive.getPe());
        setOffer_shares(receive.getOfferShares());
        setTotal_shares(receive.getTotalShares());
        setTotalAssets(receive.getTotalAssets());
        setLiquidAssets(receive.getLiquidAssets());
        setFixedAssets(receive.getFixedAssets());
        setReserved(receive.getReserved());
        setReservedPerShare(receive.getReservedPerShare());
        setEsp(receive.getEsp());
        setBvps(receive.getBvps());
        setPb(receive.getPb());
        setList_date(receive.getListDate());
        setNo_dividend(receive.getNoDividend());
        setPerundp(receive.getPerundp());
        setRev_per(receive.getRevPer());
        setProfit(receive.getProfit());
        setGross_profit(receive.getGrossProfit());
        setNet_profit(receive.getNetProfit());
        setHolder(receive.getHolder());
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
    }

    public StockBasicInfoBO_OTW(StockBasicInfoBO bofrom){
        builder = StockBasicInfoBO_PRO.StockBasicInfoBO.newBuilder();
        bo= new StockBasicInfoBO();
        setStockcode(bofrom.stockcode);
        setStockname(bofrom.stockname);
        setIndustry(bofrom.industry);
        setArea(bofrom.area);
        setPe(bofrom.pe);
        setOffer_shares(bofrom.offer_shares);
        setTotal_shares(bofrom.total_shares);
        setTotalAssets(bofrom.totalAssets);
        setLiquidAssets(bofrom.liquidAssets);
        setFixedAssets(bofrom.fixedAssets);
        setReserved(bofrom.reserved);
        setReservedPerShare(bofrom.reservedPerShare);
        setEsp(bofrom.esp);
        setBvps(bofrom.bvps);
        setPb(bofrom.pb);
        setList_date(bofrom.list_date);
        setNo_dividend(bofrom.no_dividend);
        setPerundp(bofrom.perundp);
        setRev_per(bofrom.rev_per);
        setProfit(bofrom.profit);
        setGross_profit(bofrom.gross_profit);
        setNet_profit(bofrom.net_profit);
        setHolder(bofrom.holder);
        setUuid(bofrom.uuid);
        setBoid(bofrom.boid);
        setDestination(bofrom.destination);
    }

    public byte[] serial() {
        return builder.build().toByteArray();
    }

    @Override
    public com.google.protobuf.GeneratedMessageV3.Builder getBuilder() { 
        return builder;
    }

    @Override
    public StockBasicInfoBO getBo() { 
        return bo;
    }

    public java.lang.String getStockcode() {
        return builder.getStockcode();
    }

    public void setStockcode(java.lang.String stockcode) {
        bo.stockcode = stockcode;
        builder.setStockcode(stockcode);
    }

    public java.lang.String getStockname() {
        return builder.getStockname();
    }

    public void setStockname(java.lang.String stockname) {
        bo.stockname = stockname;
        builder.setStockname(stockname);
    }

    public java.lang.String getIndustry() {
        return builder.getIndustry();
    }

    public void setIndustry(java.lang.String industry) {
        bo.industry = industry;
        builder.setIndustry(industry);
    }

    public java.lang.String getArea() {
        return builder.getArea();
    }

    public void setArea(java.lang.String area) {
        bo.area = area;
        builder.setArea(area);
    }

    public double getPe() {
        return builder.getPe();
    }

    public void setPe(double pe) {
        bo.pe = pe;
        builder.setPe(pe);
    }

    public double getOffer_shares() {
        return builder.getOfferShares();
    }

    public void setOffer_shares(double offer_shares) {
        bo.offer_shares = offer_shares;
        builder.setOfferShares(offer_shares);
    }

    public double getTotal_shares() {
        return builder.getTotalShares();
    }

    public void setTotal_shares(double total_shares) {
        bo.total_shares = total_shares;
        builder.setTotalShares(total_shares);
    }

    public double getTotalAssets() {
        return builder.getTotalAssets();
    }

    public void setTotalAssets(double totalAssets) {
        bo.totalAssets = totalAssets;
        builder.setTotalAssets(totalAssets);
    }

    public double getLiquidAssets() {
        return builder.getLiquidAssets();
    }

    public void setLiquidAssets(double liquidAssets) {
        bo.liquidAssets = liquidAssets;
        builder.setLiquidAssets(liquidAssets);
    }

    public double getFixedAssets() {
        return builder.getFixedAssets();
    }

    public void setFixedAssets(double fixedAssets) {
        bo.fixedAssets = fixedAssets;
        builder.setFixedAssets(fixedAssets);
    }

    public double getReserved() {
        return builder.getReserved();
    }

    public void setReserved(double reserved) {
        bo.reserved = reserved;
        builder.setReserved(reserved);
    }

    public double getReservedPerShare() {
        return builder.getReservedPerShare();
    }

    public void setReservedPerShare(double reservedPerShare) {
        bo.reservedPerShare = reservedPerShare;
        builder.setReservedPerShare(reservedPerShare);
    }

    public double getEsp() {
        return builder.getEsp();
    }

    public void setEsp(double esp) {
        bo.esp = esp;
        builder.setEsp(esp);
    }

    public double getBvps() {
        return builder.getBvps();
    }

    public void setBvps(double bvps) {
        bo.bvps = bvps;
        builder.setBvps(bvps);
    }

    public double getPb() {
        return builder.getPb();
    }

    public void setPb(double pb) {
        bo.pb = pb;
        builder.setPb(pb);
    }

    public java.lang.String getList_date() {
        return builder.getListDate();
    }

    public void setList_date(java.lang.String list_date) {
        bo.list_date = list_date;
        builder.setListDate(list_date);
    }

    public double getNo_dividend() {
        return builder.getNoDividend();
    }

    public void setNo_dividend(double no_dividend) {
        bo.no_dividend = no_dividend;
        builder.setNoDividend(no_dividend);
    }

    public double getPerundp() {
        return builder.getPerundp();
    }

    public void setPerundp(double perundp) {
        bo.perundp = perundp;
        builder.setPerundp(perundp);
    }

    public double getRev_per() {
        return builder.getRevPer();
    }

    public void setRev_per(double rev_per) {
        bo.rev_per = rev_per;
        builder.setRevPer(rev_per);
    }

    public double getProfit() {
        return builder.getProfit();
    }

    public void setProfit(double profit) {
        bo.profit = profit;
        builder.setProfit(profit);
    }

    public double getGross_profit() {
        return builder.getGrossProfit();
    }

    public void setGross_profit(double gross_profit) {
        bo.gross_profit = gross_profit;
        builder.setGrossProfit(gross_profit);
    }

    public double getNet_profit() {
        return builder.getNetProfit();
    }

    public void setNet_profit(double net_profit) {
        bo.net_profit = net_profit;
        builder.setNetProfit(net_profit);
    }

    public int getHolder() {
        return builder.getHolder();
    }

    public void setHolder(int holder) {
        bo.holder = holder;
        builder.setHolder(holder);
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public String toString() {
        return "StockBasicInfoBO_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "stockcode = " + getStockcode() +"," +
            "stockname = " + getStockname() +"," +
            "industry = " + getIndustry() +"," +
            "area = " + getArea() +"," +
            "pe = " + getPe() +"," +
            "offer_shares = " + getOffer_shares() +"," +
            "total_shares = " + getTotal_shares() +"," +
            "totalAssets = " + getTotalAssets() +"," +
            "liquidAssets = " + getLiquidAssets() +"," +
            "fixedAssets = " + getFixedAssets() +"," +
            "reserved = " + getReserved() +"," +
            "reservedPerShare = " + getReservedPerShare() +"," +
            "esp = " + getEsp() +"," +
            "bvps = " + getBvps() +"," +
            "pb = " + getPb() +"," +
            "list_date = " + getList_date() +"," +
            "no_dividend = " + getNo_dividend() +"," +
            "perundp = " + getPerundp() +"," +
            "rev_per = " + getRev_per() +"," +
            "profit = " + getProfit() +"," +
            "gross_profit = " + getGross_profit() +"," +
            "net_profit = " + getNet_profit() +"," +
            "holder = " + getHolder() +"," +
         "]";
    }
}
